WooCommerce/Classes/ViewRelated/Orders/Order Details/Receipts/ReceiptViewController.swift

```diff
@@ -0,0 +1,72 @@
+import WebKit
+
+/// Previews a backend-generated receipt
+final class ReceiptViewController: UIViewController {
+    @IBOutlet private weak var webView: WKWebView!
+    private let viewModel: ReceiptViewModel
+
+    init(viewModel: ReceiptViewModel) {
+        self.viewModel = viewModel
+        super.init(nibName: "LegacyReceiptViewController", bundle: nil)
+    }
+
+    required init?(coder: NSCoder) {
+        fatalError("init(coder:) has not been implemented")
+    }
+
+    override func viewDidLoad() {
+        super.viewDidLoad()
+
+        configureContent()
+        configureNavigation()
+    }
+
+    private func configureContent() {
+        guard let receipt = viewModel.receiptRequest else {
+            DDLogError("No receipt could be found.")
+            navigationController?.popViewController(animated: true)
+            return
+        }
+        webView.load(receipt)
+    }
+
+    private func configureNavigation() {
+        let shareButton = UIBarButtonItem(image: UIImage(systemName: "square.and.arrow.up"),
+                                          style: .plain,
+                                          target: self,
+                                          action: #selector(shareReceipt))
+        let printButton = UIBarButtonItem(image: UIImage(systemName: "printer"),
+                                          style: .plain,
+                                          target: self,
+                                          action: #selector(printReceipt))
+        navigationItem.rightBarButtonItems = [shareButton, printButton]
+    }
+
+    @objc private func shareReceipt() {
+        guard let url = URL(string: viewModel.receiptURLString) else {
+            return
+        }
+        let activityViewController = UIActivityViewController(activityItems: [url],
+                                                applicationActivities: nil)
+        present(activityViewController, animated: true)
+    }
+
+    @objc private func printReceipt() {
+        guard let url = URL(string: viewModel.receiptURLString) else {
+            return
+        }
+        let printController = UIPrintInteractionController.shared
+        let printInfo = UIPrintInfo(dictionary: nil)
+        printInfo.jobName = "Receipt"
```

Comment:
I'm thinking we could pass the order ID or some more info to the job name through the model, but perhaps we don't need this `jobName` at all.

Reply:
The default is the app's name, so it might be good to include that too. Given we can print shipping labels from the app, I think it would be best as "Woo: receipt for order #10289 on OsomStore", with a similar alternative for shipping labels and customs forms. These should be localized too.

The merchant can see this in the Print Center, [accessed from the app switcher while a print is ongoing (TIL)](https://support.apple.com/en-us/109349)

![Print Center screenshot](https://github.com/woocommerce/woocommerce-ios/assets/2472348/e6506ad3-731b-4ca5-86cb-9ba8b7797599)